crates/junction-api/src/xds/http.rs

```diff
@@ -70,9 +87,45 @@ impl Route {
                         .with_field_index("virtual_hosts", idx);
                 };
 
+                let vhost_ratelimit: xds_local_ratelimit::LocalRateLimit = vhost
+                    .typed_per_filter_config
+                    .get("envoy.filters.http.local_ratelimit")
+                    .map(|any| {
+                        protobuf::Any::to_msg(any).or(Err(Error::new_static(
+                            "Failed to parse local rate limit config",
+                        )
+                        .with_field("envoy.filters.http.local.ratelimit")))
+                    })
+                    .transpose()
+                    .and_then(|r| r.ok_or(Error::new_static("missing local rate limit config")))
+                    .with_field("typed_per_filter_config")
+                    .with_field_index("virtual_hosts", idx)?;
+
+                let route = vhost
```

Comment:
I thiiiiiiiink it might be easier to shove the `typed_per_filter_config` into `actions_and_matches` (and to rename `actions_and_matches`). The intent of that section is to do all of the field walking and group things ahead of time, so you can think in Junction shapes instead of xDS shapes. 

```rust
            let route_stuff  = vhost.routes.iter().enumerate().map(|(route_idx, route)| {
                 let action = route.action.as_ref();
                 let route_parts = (route_idx, route.r#match.as_ref(), &route.typed_per_filter_config);

                 (action, route_parts)
            });
```

It might make that tuple a little extra gnarly, but try it out and see.